discovery-provider/src/tasks/index.py

```diff
@@ -122,6 +125,9 @@ def index_blocks(self, db, blocks_list):
             former_current_block.is_current = False
             session.add(block_model)
 
+            # add the current block to redis
+            redis.set(most_recent_indexed_block_redis_key, block.number)
```

Comment:
this should be _after_ indexing for this block is done, below the sorted_txs

Reply:
@hareeshnagaraj should be fixed now